### ISSUE TITLE
Add an "always"/"never" option for wrap-regex

### ISSUE THREAD

**Issue by zypA13510** [eslint/eslint#10385]
**What rule do you want to change?**
wrap-regex
As of now (ESLint version 4.19.1?), `no-extra-parens` does not handle RegExp literals, and `wrap-regex` allows only "always" behavior, but not a "never" behavior (i.e. never wrap RegExp literals with parentheses). So I propose that an "always"/"never" option added to the rule `wrap-regex`.

**Does this change cause the rule to produce more or fewer warnings?**
With the option set to "always", it would produce the same amount of warnings.
With the option set to "never", it would produce more warnings.

**How will the change be implemented? (New option, new default behavior, etc.)?**
New option. Defaults to "always" to remain backward compatible.

### What does the rule currently do for this code?
Example of **incorrect** code for this rule:
```js
/*eslint wrap-regex: "error"*/

function a() {
    return /foo/.test("bar");
}
```
Example of **correct** code for this rule:
```js
/*eslint wrap-regex: "error"*/

function a() {
    return (/foo/).test("bar");
}
```
### What will the rule do after it's changed?
Example of **incorrect** code for the default `"always"` option:
```js
/*eslint wrap-regex: ["error", "always"]*/

function a() {
    return /foo/.test("bar");
}
```
Example of **correct** code for the default `"always"` option:
```js
/*eslint wrap-regex: ["error", "always"]*/

function a() {
    return (/foo/).test("bar");
}
```
Example of **incorrect** code for the `"never"` option:
```js
/*eslint wrap-regex: ["error", "never"]*/

function a() {
    return (/foo/).test("bar");
}
```
Example of **correct** code for the `"never"` option:
```js
/*eslint wrap-regex: ["error", "never"]*/

function a() {
    return /foo/.test("bar");
}
```

**Comment by not-an-aardvark** [comment]
Thanks for creating an issue and for your interest in improving ESLint. Unfortunately, it looks like this issue didn't get consensus from the team, so I'm closing it. We define [consensus](http://eslint.org/docs/maintainer-guide/issues.html#consensus) as having three :+1:s from team members, as well as a team member willing to champion the proposal. This is a high bar by design -- we can't realistically accept and maintain every feature request in the long term, so we only accept feature requests which are useful enough that there is consensus among the team that they're worth adding.

Since ESLint is pluggable and can load custom rules at runtime, the lack of consensus among the ESLint team doesn't need to be a blocker for you using this in your project, if you'd find it useful. It just means that you would need to [implement the rule](http://eslint.org/docs/developer-guide/working-with-rules) yourself, rather than using a bundled rule that is packaged with ESLint.
